tokio-signal/examples/ctrl-c.rs

```diff
@@ -31,22 +31,20 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
     // Stream::for_each is a powerful primitive provided by the Futures crate.
     // It turns a Stream into a Future that completes after all stream-items
     // have been completed, or the first time the closure returns an error
-    let future = limited_stream
-        .map(|result| result.expect("failed to get event"))
-        .for_each(|()| {
-            // Note how we manipulate the counter without any fancy synchronisation.
-            // The borrowchecker realises there can't be any conflicts, so the closure
-            // can just capture it.
-            counter += 1;
-            println!(
-                "Ctrl+C received {} times! {} more before exit",
-                counter,
-                STOP_AFTER - counter
-            );
-
-            // return a result to continue handling the stream
-            future::ready(())
-        });
+    let future = limited_stream.for_each(|()| {
```

Comment:
maybe it'd be nice to use a `while let`?

Reply:
At first I omitted using `while let` due to `await` not working on streams, but it appears that we can use `stream.next().await` which looks much nicer, so I'll update this!